controllers/util/provider.go

```diff
@@ -19,15 +19,20 @@ import (
 
 // GenerateOnboardingToken generates a token valid for a duration of "tokenLifetimeInHours".
 // The token content is predefined and signed by the private key which'll be read from supplied "privateKeyPath".
-func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string) (string, error) {
+// The storageQuotaInGiB is optional, and it is used to limit the storage of PVC in the application cluster.
+func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string, storageQuotaInGiB *int) (string, error) {
 	tokenExpirationDate := time.Now().
 		Add(time.Duration(tokenLifetimeInHours) * time.Hour).
 		Unix()
 
-	payload, err := json.Marshal(services.OnboardingTicket{
+	ticket := services.OnboardingTicket{
 		ID:             uuid.New().String(),
 		ExpirationDate: tokenExpirationDate,
-	})
+	}
+	if storageQuotaInGiB != nil {
+		ticket.StorageQuotaInGiB = *storageQuotaInGiB
```

Comment:
@rchikatw You decided to use an *int as opposed to *uint. You need to validate that the passed value is greater than 0 to eliminate cases where someone invoked with 
0 or a negative number

Reply:
I am validating against negative numbers here https://github.com/red-hat-storage/ocs-operator/pull/2600/files#diff-5575cae66180001e0066d01f86d13ae4a7137f5424f09f4af76abfefc5eb8dd7R46 but using `*unit` no need to add this condition.

Reply:
Quota need to be validate to be greater then 0 which is not forced by uint